packages/@lwc/synthetic-shadow/src/shared/utils.ts

```diff
@@ -51,7 +51,7 @@ export function arrayFromCollection<T extends Node, K extends Element>(
     collection: NodeListOf<T> | HTMLCollectionOf<K>
 ): Array<T | K> {
     const size = collection.length;
-    const cloned: T[] | K[] = new ArrayConstructor(size);
```

Comment:
Changed the array instantiation to an array literal. 

On top of being better understood by typescript, it is also better optimized by javascript engines in general. Creating an array with a fixed size has the side effect to confuse the JIT optimizer in most of the engines.

Reply:
hmm, there was a reason for this... and I think it was performance.... anyhow...

Reply:
The code becomes way slower if the Array constructor is dereferenced: https://jsperf.com/array-constructor-vs-literal-set.